addons/website_slides/static/src/js/slides_course_fullscreen_player.js

```diff
@@ -7,413 +7,585 @@ odoo.define('website_slides.fullscreen', function (require) {
-            self.player = undefined;
-            self.goToQuiz = self.slide.hasQuiz && !self.goToQuiz && self.slide.slide_type !== 'quiz';
-            if (self.nextSlide && !self.goToQuiz){
-                self.slide = self.nextSlide;
-                self.index++;
-                self._setActiveTab();
-                self._renderPlayer();
-                self._setPreviousAndNextSlides();
-                self._updateUrl();
-                history.pushState(null,'',self.url);
-            }
-            else if (self.nextSlide){
-                self._renderPlayer();
-            }
-        },
-        _goToPreviousSlide: function (){
-            var self = this;
-            clearInterval(self.tid);
-            self.goToQuiz = false;
-            self.player = undefined;
-            if (self.previousSlide){
-                self.slide = self.previousSlide;
-                self.index--;
-                self._setActiveTab();
-                self._renderPlayer();
-                self._setPreviousAndNextSlides();
-                self._updateUrl();
-                history.pushState(null,'',self.url);
-            }
+        _findSlide: function (matcher) {
+            var slideMatch = _.matcher(matcher);
+            var result = _.filter(this.slides, slideMatch);
+            return result[0];
         },
-        _setPreviousAndNextSlides: function (){
-            var self = this;
-            self.previousSlide = self.index > 0 ? self.slides[self.index-1] : undefined;
-            self.nextSlide = self.index < (self.slides.length - 1) ? self.slides[self.index+1] : undefined;
+        /**
+         * Extend the slide data list to add informations about rendering method, and other
+         * specific values according to their slide_type.
+         */
+        _preprocessSlideData: function(slidesDataList) {
+            _.each(slidesDataList, function (slideData, index) {
+                // compute hasNext slide
+                slideData.hasNext = index < slidesDataList.length-1;
+                // compute embed url
+                if (slideData.type === 'video') {
+                    slideData.embedCode = $(slideData.embedCode).attr('src');  // embedCode containts an iframe tag, where src attribute is the url (youtube or embed document from odoo)
+                    slideData.embedUrl =  "https://" + slideData.embedCode + "&rel=0&autoplay=1&enablejsapi=1&origin=" + window.location.origin;
+                } else if (slideData.type === 'infographic') {
+                    slideData.embedUrl = _.str.sprintf('/web/image/slide.slide/%s/image', slideData.id);
+                } else if (_.contains(['document', 'presentation'], slideData.type)) {
+                    slideData.embedUrl = $(slideData.embedCode).attr('src');
+                }
+                // fill empty property to allow searching on it with _.filter(list, matcher)
+                slideData.isQuiz = slideData.isQuiz ? true : false;
```

Comment:
That looks a little weird.
Can't you use `=== true` at least?

Reply:
The API of `.data()` is strange. When the value is evaluated to false, the attribute does not exists on the html node. `t-att-data-caca="0"` will return `undefined` on `node.data('caca')`. 
We need here to have explicit values in dict (for the matcher). This condition is strange, maybe `                slideData.isQuiz = !!slideData.isQuiz;` is less weird ?

Reply:
Then maybe `pyUtils.py_eval` would be the thing to use here?